api/types/license_test.go

```diff
@@ -0,0 +1,150 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package types
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestLicenseSettersAndGetters(t *testing.T) {
+	tt := []struct {
+		name        string
+		setter      func(License, Bool)
+		getter      func(License) Bool
+		unsetValues [](func(License) Bool)
+	}{
+		{
+			name:   "Set ReportsUsage",
+			setter: License.SetReportsUsage,
+			getter: License.GetReportsUsage,
+			unsetValues: []func(License) Bool{
```

Comment:
Could we have a single list with all license getters and have the test skip the current getter from it? It's easier to review a single list than having to eyeball the diffs in every test case.